packages/smooth_app/lib/pages/product/edit_ocr_page.dart

```diff
@@ -377,4 +375,224 @@ class _EditOcrPageState extends State<EditOcrPage> with UpToDateMixin {
+      OcrState.EXTRACTING_DATA => _EditOcrActionExtractingContent(
+          helper: helper,
+          appLocalizations: appLocalizations,
+        ),
+      OcrState.OTHER => null,
+    };
+
+    if (child == null) {
+      return EMPTY_WIDGET;
+    }
+
+    final SmoothColorsThemeExtension theme =
+        Theme.of(context).extension<SmoothColorsThemeExtension>()!;
+
+    return SizedBox(
+      height: 45.0 * (_computeFontScaleFactor(context)),
+      width: double.infinity,
+      child: DecoratedBox(
+        decoration: BoxDecoration(
+          borderRadius: ANGULAR_BORDER_RADIUS,
+          color: theme.primarySemiDark,
+          border: Border.all(
+            color: theme.primaryBlack,
+            width: 2.0,
+          ),
+        ),
+        child: Material(
+          type: MaterialType.transparency,
+          child: ProgressIndicatorTheme(
+            data: const ProgressIndicatorThemeData(
+              color: Colors.white,
+            ),
+            child: IconTheme(
+              data: const IconThemeData(color: Colors.white),
+              child: DefaultTextStyle(
+                style: const TextStyle(
+                  fontWeight: FontWeight.bold,
+                  fontSize: 15.5,
+                  color: Colors.white,
+                ),
+                child: child,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  double _computeFontScaleFactor(BuildContext context) {
+    final double fontSize = DefaultTextStyle.of(context).style.fontSize ?? 15.0;
+    final double scaledFontSize =
+        MediaQuery.textScalerOf(context).scale(fontSize);
+
+    return scaledFontSize / fontSize;
+  }
+}
+
+class _EditOcrActionExtractingContent extends StatelessWidget {
+  const _EditOcrActionExtractingContent({
+    required this.helper,
+    required this.appLocalizations,
+  });
+
+  final OcrHelper helper;
+  final AppLocalizations appLocalizations;
+
+  @override
+  Widget build(BuildContext context) {
+    return Semantics(
+      label: helper.getActionLoadingPhoto(appLocalizations),
+      excludeSemantics: true,
+      child: Shimmer(
+        gradient: const LinearGradient(
+          begin: Alignment.centerLeft,
+          end: Alignment.centerRight,
+          colors: <Color>[
+            Colors.black,
+            Colors.white,
+            Colors.black,
+          ],
+        ),
+        child: Padding(
+          padding: const EdgeInsets.symmetric(horizontal: MEDIUM_SPACE),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: <Widget>[
+              const CircularProgressIndicator.adaptive(),
+              Expanded(
+                child: Text(
+                  helper.getActionExtractingData(appLocalizations),
+                  textAlign: TextAlign.center,
+                ),
+              ),
+              const CircularProgressIndicator.adaptive(),
```

Comment:
2 progress indicators? A bit too much, isn't it?

Reply:
You can see it, on the video, with the Shimmer effect, actually it's OK.
To be honest, I've not checked on Android and in that case, I will remove the first one on non-Apple devices